blocks/api/factory.js

```diff
@@ -116,3 +121,19 @@ export function switchToBlockType( block, name ) {
 		uid: index === firstSwitchedBlock ? block.uid : result.uid,
 	} ) );
 }
+
+/**
+ * Creates a new reusable block.
+ *
+ * @param {String} type       The type of the block referenced by the reusable block
+ * @param {Object} attributes The attributes of the block referenced by the reusable block
+ * @return {Object}           A reusable block object
+ */
+export function createReusableBlock( type, attributes ) {
+	return {
+		id: uuid(),
+		name: __( 'Untitled block' ),
```

Comment:
Would there ever be a case where we want to create a reusable block and assign its name immediately? A bit awkward to achieve now, since this is not accepted as an argument of the function.

Reply:
No such case was in our design so I figured _you aren't gonna need it_. We could add an optional argument in the future if it comes up.

Reply:
I think we should support this and have UI that focus on the name input before saving. cc @jasmussen